elfi/methods/utils.py

```diff
@@ -184,29 +184,72 @@ def logpdf(cls, x, means, cov=1, weights=None):
         return np.log(cls.pdf(x, means=means, cov=cov, weights=weights))
 
     @classmethod
-    def rvs(cls, means, cov=1, weights=None, size=1, random_state=None):
+    def rvs(cls, means, cov=1, weights=None, size=1, prior_logpdf=None,
+            max_trials=100, random_state=None):
         """Draw random variates from the distribution.
 
         Parameters
         ----------
         means : array_like
             means of the Gaussian mixture components
-        weights : array_like
-            1d array of weights of the gaussian mixture components
-        cov : array_like
+        cov : array_like, optional
             a shared covariance matrix for the mixture components
-        size : int or tuple
+        weights : array_like, optional
+            1d array of weights of the gaussian mixture components
+        size : int or tuple, optional
+        prior_logpdf : callable, optional
+            Can be used to check validity of random variable.
+        max_trials : int, optional
+            Maximum number of trials to find allowed random variables.
         random_state : np.random.RandomState or None
 
         """
-        means, weights = cls._normalize_params(means, weights)
         random_state = random_state or np.random
+        means, weights = cls._normalize_params(means, weights)
 
-        inds = random_state.choice(len(means), size=size, p=weights)
-        rvs = means[inds]
-        perturb = ss.multivariate_normal.rvs(
-            mean=means[0] * 0, cov=cov, random_state=random_state, size=size)
-        return rvs + perturb
+        # additional complexity due to supporting scipy-like ambiguity of arguments
+        if size is None:
+            size = means.shape[0]
+            output = []
+            scalar = True
+        else:
+            output = np.empty((size,) + means.shape[1:])
+            scalar = False
+
+        n_accepted = 0
+        n_left = size
+        trials = 0
+
+        while n_accepted < size:
+            inds = random_state.choice(len(means), size=n_left, p=weights)
+            rvs = means[inds]
+            perturb = ss.multivariate_normal.rvs(mean=means[0]*0,
```

Comment:
make lint

Reply:
Which errors did you get? I only get 1 `isort` problem, which seems to originate from `dev` (will fix).

Reply:
2     E226 missing whitespace around arithmetic operator

Reply:
Hmm... Strange that I don't get that error. Well, fixed.

Reply:
`pip list |grep flake8` returns
flake8 (3.4.1)
flake8-docstrings (1.1.0)
flake8-isort (2.2.2)
flake8-polyfill (1.0.1)
If these are the same, maybe you have a local ignore list?

Reply:
These are the same, but maybe you have `flake8` extensions that are named in some more arbitrary format.

`>> flake8 --help`
Installed plugins: flake8-docstrings: 1.1.0, pydocstyle: 2.0.0, flake8_isort:
2.1.3, mccabe: 0.6.1, pycodestyle: 2.3.1, pyflakes: 1.5.0

Reply:
Mine is:
Installed plugins: flake8-docstrings: 1.1.0, pydocstyle: 2.0.0, flake8_isort:
2.1.3, mccabe: 0.6.1, naming: 0.4.1, pycodestyle: 2.3.1, pyflakes: 1.5.0

Though I believe this is related to http://pep8.readthedocs.io/en/release-1.7.x/intro.html, E226 is marked with a star.